src/Adapter/ApiAccess/CommandHandler/AddApiAccessHandler.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\ApiAccess\CommandHandler;
+
+use Doctrine\ORM\Exception\ORMException;
+use PrestaShop\PrestaShop\Core\CommandBus\Attributes\AsCommandHandler;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\Command\AddApiAccessCommand;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\CommandHandler\AddApiAccessCommandHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\Exception\ApiAccessConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\Exception\CannotAddApiAccessException;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\ValueObject\ApiAccessId;
+use PrestaShopBundle\Entity\ApiAccess;
+use PrestaShopBundle\Entity\Repository\ApiAccessRepository;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+#[AsCommandHandler]
+class AddApiAccessHandler implements AddApiAccessCommandHandlerInterface
+{
+    public function __construct(
+        private readonly ApiAccessRepository $repository,
+        private readonly ValidatorInterface $validator
+    ) {
+    }
+
+    public function handle(AddApiAccessCommand $command): ApiAccessId
+    {
+        $apiAccess = new ApiAccess();
+        $apiAccess->setClientId($command->getApiClientId());
+        $apiAccess->setClientName($command->getClientName());
+        $apiAccess->setEnabled($command->isEnabled());
+        $apiAccess->setDescription($command->getDescription());
+
+        $errors = $this->validator->validate($apiAccess);
+
+        if (count($errors) > 0) {
+            if (!($errors->get(0)->getPropertyPath() === 'enabled' && $errors->get(0)->getInvalidValue() === false)) {
```

Comment:
Why do you need this special condition for enabled? All fields should be handled the same way no? If there's an error returned by the validator it should mean something is wrong